lib/msal-core/README.md

```diff
@@ -25,87 +22,117 @@ Via CDN:
+In MSAL, you can get access tokens for the APIs your app needs to call using the `acquireTokenSilent` method which makes a silent request(without prompting the user with UI) to Azure AD to obtain an access token. The Azure AD service then returns an [access token](https://docs.microsoft.com/en-us/azure/active-directory/develop/access-tokens) containing the user consented scopes to allow your app to securely call the API.
 
-If the silent token acquisition fails for some reasons such as an expired token or password change, you will need to invoke an interactive method to acquire tokens such as acquireTokenPopup or acquireTokenRedirect.
+If the `acquireTokenSilent` call fails with an error of type `InteractionRequiredAuthError` (requiring the user to consent to something), you will need to initiate an interactive request. You can use `acquireTokenRedirect` or `acquireTokenPopup`.
 
- ```JavaScript
-    var graphScopes = ["user.read", "mail.send"];
+`acquireTokenSilent` will look for a valid token in the cache, and if it is close to expiring or does not exist, will automatically refresh it for you.
 
-    userAgentApplication.loginPopup(graphScopes).then(function (idToken) {
-        //Login Success
-        userAgentApplication.acquireTokenSilent(graphScopes).then(function (accessToken) {
-            //AcquireTokenSilent Success
-        }, function (error) {
-            //AcquireTokenSilent Failure, send an interactive request.
-            userAgentApplication.acquireTokenPopup(graphScopes).then(function (accessToken) {
-                updateUI();
-            }, function (error) {
-                console.log(error);
+```JavaScript
+    // if the user is already logged in you can acquire a token
+    if (msalInstance.getAccount()) {
+        var tokenRequest = {
+            scopes: ["user.read", "mail.send"]
+        };
+        msalInstance.acquireTokenSilent(tokenRequest)
+            .then(response => {
+                // get access token from response
+                // response.accessToken
+            })
+            .catch(err => {
+                // could also check if err instance of InteractionRequiredAuthError if you can import the class.
+                if (err.name === "InteractionRequiredAuthError") {
+                    return msalnstance.acquireTokenPopup(tokenRequest)
+                        .then(response => {
+                            // get access token from response
+                            // response.accessToken
+                        })
+                        .catch(err => {
+                            // handle error
+                        });
+                }
             });
-        })
-    }, function (error) {
-        //login failure
-        console.log(error);
-    });
+    } else {
+        // user is not logged in, you will need to log them in to acquire a token
+    }
```

Comment:
Add a link to how 'response' looks like at some place.

Reply:
adding